api/applications/serializers.py

```diff
@@ -8,31 +8,34 @@
 
 class ApiOAuth2ApplicationSerializer(JSONAPISerializer):
     """Serialize data about a registered OAuth2 application"""
-    client_id = ser.CharField(help_text="The client ID for this application (automatically generated)",
+    id = ser.CharField(help_text='The client ID for this application (automatically generated)',
+                       read_only=True,
```

Comment:
Minor, but you can pass `label="ID"` so that the label is correct in `OPTIONS` requests and the forms in the  browse-able API display correctly.